src/ResourceManager/RecoveryServices.SiteRecovery/Commands.RecoveryServices.SiteRecovery/VCenter/RemoveAzureRmRecoveryServicesAsrVCenter.cs

```diff
@@ -0,0 +1,92 @@
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.RecoveryServices.SiteRecovery
+{
+    /// <summary>
+    ///     Removes the Azure Site Recovery vCenter.
+    /// </summary>
+    [Cmdlet(
+        VerbsCommon.Remove,
+        "AzureRmRecoveryServicesAsrvCenter",
+        DefaultParameterSetName = ASRParameterSets.Default,
+        SupportsShouldProcess = true)]
+    [Alias("Remove-ASRvCenter")]
+    [OutputType(typeof(ASRJob))]
+    public class RemoveAzureRmRecoveryServicesAsrvCenter : SiteRecoveryCmdletBase
+    {
+        /// <summary>
+        ///     Gets or sets Resource Id.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.ByResourceId,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true)]
+        public string ResourceId { get; set; }
+
+        /// <summary>
+        ///     Gets or sets the vCenter.
+        /// </summary>
+        [Parameter(
+            ParameterSetName = ASRParameterSets.Default,
+            Mandatory = true,
+            ValueFromPipeline = true)]
+        [ValidateNotNullOrEmpty]
+        [Alias("vCenter")]
+        public ASRvCenter InputObject { get; set; }
+
+        /// <summary>
+        ///     ProcessRecord of the command.
+        /// </summary>
+        public override void ExecuteSiteRecoveryCmdlet()
+        {
+            base.ExecuteSiteRecoveryCmdlet();
+
+            string vCenterName = null;
+            string fabricName = null;
+            switch (this.ParameterSetName)
+            {
+                case ASRParameterSets.ByResourceId:
+                    vCenterName = Utilities.GetValueFromArmId(
+                    this.ResourceId,
+                    ARMResourceTypeConstants.vCenters);
+
+                    fabricName = Utilities.GetValueFromArmId(
+                    this.ResourceId,
+                    ARMResourceTypeConstants.ReplicationFabrics);
+                    break;
+
+                case ASRParameterSets.Default:
+                    vCenterName = this.InputObject.Name;
+                    fabricName = this.InputObject.FabricArmResourceName;
+                    break;
+            }
+
+            if (this.ShouldProcess(vCenterName, VerbsCommon.Remove))
+            {
+                var response = this.RecoveryServicesClient.RemoveAzureRmSiteRecoveryvCenter(
+                    fabricName,
```

Comment:
@viverm just a general question for cmdlets that have a `Fabric` parameter (and this might have been discussed already): why don't we make the user provide a `FabricName` parameter instead of the entire object? That way, they don't have to get the `Fabric` object each time they want to run these cmdlets and can just provide a string instead. Is the `FabricName` hard to discover for the user, or difficult to remember?

Reply:
Fabric Name  is ASR generated entity which is unique .  It is Difficult for user to remember and pass it. user will store and pass it so he can fabric object like he is doing in other cmdlets.